addons/website_sale/models/res_config_settings.py

```diff
@@ -81,19 +81,24 @@ def get_values(self):
 
         res.update(
             sale_delivery_settings=sale_delivery_settings,
-            enabled_extra_checkout_step=self.env.ref('website_sale.extra_info_option').active,
-            enabled_buy_now_button=self.env.ref('website_sale.product_buy_now').active,
         )
         return res
 
     def set_values(self):
         super().set_values()
-        extra_step_view = self.env.ref('website_sale.extra_info_option')
-        if extra_step_view.active != self.enabled_extra_checkout_step:
-            extra_step_view.active = self.enabled_extra_checkout_step
-        buy_now_view = self.env.ref('website_sale.product_buy_now')
-        if buy_now_view.active != self.enabled_buy_now_button:
-            buy_now_view.active = self.enabled_buy_now_button
+        if self.website_id:
+            extra_step_views = self.with_context(website_id=self.website_id.id).website_id.viewref(
+                'website_sale.extra_info_option')
+            buy_now_views = self.with_context(website_id=self.website_id.id).website_id.viewref(
+                'website_sale.product_buy_now')
+
+            for extra_step_view in extra_step_views:
+                if extra_step_view.active != self.enabled_extra_checkout_step:
+                    extra_step_view.active = self.enabled_extra_checkout_step
+
+            for buy_now_view in buy_now_views:
+                if buy_now_view.active != self.enabled_buy_now_button:
+                    buy_now_view.active = self.enabled_buy_now_button
```

Comment:
No need for loops :)
```suggestion
            extra_step_views.active = self.enabled_extra_checkout_step
            buy_now_views.active = self.enabled_buy_now_button
```